test/token_holder/execute_rule.js

```diff
@@ -0,0 +1,605 @@
+                {
+                    type: 'address',
+                    name: 'value',
+                },
+            ],
+        },
+        [value],
+    );
+}
+
+function generateExecuteRuleCallPrefix() {
+    return web3.eth.abi.encodeFunctionSignature({
+        name: 'executeRule',
+        type: 'function',
+        inputs: [
+            {
+                type: 'address', name: '',
+            },
+            {
+                type: 'bytes', name: '',
+            },
+            {
+                type: 'uint256', name: '',
+            },
+            {
+                type: 'uint8', name: '',
+            },
+            {
+                type: 'bytes32', name: '',
+            },
+            {
+                type: 'bytes32', name: '',
+            },
+        ],
+    });
+}
+
+function getExecuteRuleExTxHash(
+    tokenHolderAddress, ruleAddress, ruleData, nonce,
+) {
+    return web3.utils.soliditySha3(
+        {
+            t: 'bytes1', v: '0x19',
+        },
+        {
+            t: 'bytes1', v: '0x0',
+        },
+        {
+            t: 'address', v: tokenHolderAddress,
+        },
+        {
+            t: 'address', v: ruleAddress,
+        },
+        {
+            t: 'uint8', v: 0,
+        },
+        {
+            t: 'bytes32', v: web3.utils.keccak256(ruleData),
+        },
+        {
+            t: 'uint256', v: nonce,
+        },
+        {
+            t: 'uint8', v: 0,
+        },
+        {
+            t: 'uint8', v: 0,
+        },
+        {
+            t: 'uint8', v: 0,
+        },
+        {
+            t: 'bytes4', v: generateExecuteRuleCallPrefix(),
+        },
+        {
+            t: 'uint8', v: 0,
+        },
+        {
+            t: 'bytes32', v: '0x0',
+        },
+    );
+}
+
+function getExecuteRuleExTxData(
+    _tokenHolderAddress, _ruleAddress, _ruleData, _nonce, _ephemeralKey,
+) {
+    const msgHash = getExecuteRuleExTxHash(
+        _tokenHolderAddress, _ruleAddress, _ruleData, _nonce,
+    );
+
+    const rsv = ethUtils.ecsign(
+        ethUtils.toBuffer(msgHash),
+        ethUtils.toBuffer(_ephemeralKey),
+    );
+
+    return [msgHash, rsv];
+}
+
+async function createTokenHolder(
+    _wallet, _ephemeralKeyAddress, _spendingLimit, _deltaExpirationHeight,
+) {
+    const token = await EIP20TokenMock.new(1, 1, 'OST', 'Open Simple Token', 1);
+    const tokenRules = await TokenRulesMock.new();
+    const required = 1;
+    const registeredWallet0 = _wallet;
+    const wallets = [registeredWallet0];
+
+    const tokenHolder = await TokenHolder.new(
+        token.address,
+        tokenRules.address,
+        required,
+        wallets,
+    );
+
+    await tokenHolder.submitAuthorizeSession(
+        _ephemeralKeyAddress,
+        _spendingLimit,
+        (await web3.eth.getBlockNumber()) + _deltaExpirationHeight,
+        {
+            from: registeredWallet0,
+        },
+    );
+
+    return tokenHolder;
+}
+
+contract('TokenHolder::executeRule', async () => {
```

Comment:
Add test case where ephemeralkey to be authorized is passed null and empty.

Reply:
This is a unit test for TokenHolder::executeRule.
Are you asking to test TokenHolder::authorizeSession function here also, or?
If it's about testing TokenHolder::authorizeSession then it will be done in separate test (already implemented in PR #50).